### ISSUE TITLE
Define criteria for bet sizes

### ISSUE THREAD

**Issue by kongzii** [gnosis/prediction-market-agent#56]
Gabriel's original text:

Having worked a bit more in the PMA (https://github.com/gnosis/prediction-market-agent) repo, I believe we should think about having some infrastructure for the agent being able to determine the size of each bet. Currently this is the (standard) implementation for this:

```
def calculate_bet_amount(self, answer: bool, market: AgentMarket) -> BetAmount:
    """
    Calculate the bet amount. By default, it returns the minimum bet amount.
    """
    return market.get_tiny_bet_amount()
```

One option that comes to mind is let the agent determine the amount, based on a number of factors, e.g. xDAI balance, confidence on prediction, time left for market to close, etc. Those are all parameters we can already provide the agent.

**Comment by kongzii** [comment]
Moving https://github.com/gnosis/prediction-market-agent-tooling/issues/157 into PMA, to keep in mind that some more intelligent bet sizing is desirable.

If any universal functions/classes/methods come out of implementing this, we can promote that to the PMAT repo, of course.

**Comment by kongzii** [comment]
Currently blocked by https://github.com/gnosis/prediction-market-agent-tooling/issues/161

**Comment by gabrielfior** [comment]
Another suggestion :
-> If we are defining bet_sizes in a mathematical way (e.g. based on difference between `agent_p_yes` and `market_p_yes`), then it's OK to have `bet_amount : float`
-> If we let the LLM decide on the bet amount, probably better to have it pick from a set of fixed sizes (e.g. `pick from [0.01, 0.1, 1., 10] xDAI instead of allowing for a continous range (see https://arize.com/blog-course/numeric-evals-for-llm-as-a-judge/)

**Comment by evangriffiths** [comment]
We do have these betting strategies implemented in python to choose from https://github.com/gnosis/prediction-market-agent-tooling/tree/main/prediction_market_agent_tooling/tools/betting_strategies

**Comment by evangriffiths** [comment]
Closing, as duplicate of https://github.com/gnosis/prediction-market-agent/issues/280
